src/metadataGeneration/metadataGenerator.ts

```diff
@@ -56,6 +56,34 @@ export class MetadataGenerator {
       }
 
       ts.forEachChild(sf, node => {
+        /**
+         * If we declare a namespace within a module, like we do in `tsoaTestModule.d.ts`,
+         * we need to explicitly get the children of the module declaration
+         * (`declare module 'tsoaTest'`) - which are the moduleBlock statements,
+         * because otherwise our type resolver cannot iterate over namespaces defined in that module.
+         */
+        if (ts.isModuleDeclaration(node)) {
+          const moduleDeclaration = node as ts.ModuleDeclaration;
+          /**
+           * For some reason unknown to me, TS resolves both `declare module` and `namespace` to
+           * the same kind (`ModuleDeclaration`). In order to figure out whether it's one or the other,
+           * we check the name property. When it is a string literal, we assume that it's a
+           * module declaration, not a namespace. I haven't found any other differences between the
+           * objects, except `name` and `localSymbol`. Since `localSymbol` is not defined anywhere in
+           * the types, we fall back on the `name` checking.
+           */
+          if (moduleDeclaration.name.kind === ts.SyntaxKind.StringLiteral) {
```

Comment:
you can check based on the flags so we can avoid this

Reply:
Fixed.